### ISSUE TITLE
Release SWE-World artifacts (models, dataset) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [RUCAIBox/SWE-World#2]
Hello,

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2602.03419.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I saw in your GitHub README that the code, the **SWT** and **SWR** models, and the **SWE-World Dataset** are currently undergoing internal corporate review for release. It'd be great to make these checkpoints and the dataset available on the 🤗 hub once they are cleared, to improve their discoverability and visibility within the software engineering and AI communities.

We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, for the LLM-based transition and reward models, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverage the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint (like SWT-72B and SWR-72B) to a separate model repository so that download stats work individually.

## Uploading dataset

Would be awesome to make the SWE-World dataset available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/swe-world")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the repositories and tasks in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by DeepLSUN** [comment]
We have open-sourced the related models at  
https://huggingface.co/collections/RUC-AIBOX/swe-agent-series.

Due to company restrictions, the datasets cannot be publicly released at this time.  
However, we have open-sourced the data processing pipeline in this repository.

Thank you for your support.

**Comment by NielsRogge** [comment]
Hey @DeepLSUN, it's great to see that you've uploaded the models to the Hugging Face hub.

However, it seems they are not yet linked to the paper page. Could you link them by mentioning the paper URL (huggingface.co/papers/2602.03419) in the model cards?
This will make the models automatically show up on the paper page.

See also here for more information: https://huggingface.co/docs/hub/model-release-checklist#write-a-comprehensive-model-card

Thanks
